dashi/src/lib/components/DashiPlot.tsx

```diff
@@ -16,23 +22,27 @@ export function DashiPlot({
   if (!figure) {
     return <div id={id} style={style} />;
   }
-  return (
-    <Plot
-      divId={id}
-      style={style}
-      data={figure.data}
-      layout={figure.layout}
-      config={figure.config}
-      onClick={(event) => {
-        if (id) {
-          onPropertyChange({
-            componentType: "Plot",
-            componentId: id,
-            propertyName: "points",
-            propertyValue: event.points,
-          });
-        }
-      }}
-    />
-  );
+  const { datasets, ...spec } = figure;
+  const Plot = createClassFromSpec({
+    mode: "vega-lite",
+    spec: spec,
+  });
```

Comment:
I believe, we should use [approach #3](https://github.com/vega/react-vega/tree/master/packages/react-vega#approach3-use-vegalite-generic-class-and-pass-in-spec-for-dynamic-component), namely using `<VegaLite>` generic class and pass in spec for dynamic component. Creating the class every time the spec changes is very inefficient. We want the instantiated component to be (re)rendered every time the spec changes.